connectors/http/http-base/src/main/java/io/camunda/connector/http/base/blocklist/block/PortBlock.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * <p>The Port Block is defined by a comma-separated list of port numbers and a block name. If a
+ * URL's port matches any of the specified port numbers, it is considered blocked, and an exception
+ * is thrown.
+ */
+public final class PortBlock implements Block {
+  private final String value;
+  private final String blockName;
+
+  private final Set<Integer> blockedPorts;
+
+  /**
+   * Creates a new instance of the PortBlock class with the specified port numbers and block name.
+   *
+   * @param value The comma-separated list of port numbers used for blocking.
+   * @param blockName The name of the block for identification.
+   * @throws IllegalArgumentException if the provided port numbers or format are invalid.
+   */
+  public PortBlock(final String value, final String blockName) {
+    this.value = value;
+    this.blockName = blockName;
+    this.blockedPorts = validateAndSetPorts(value);
+  }
+
+  private Set<Integer> validateAndSetPorts(String portValue) {
+    String[] ports = portValue.split(",");
+    Set<Integer> parsedPorts = new HashSet<>();
+    for (String port : ports) {
+      try {
+        int parsedPort = Integer.parseInt(port.trim());
+        if (parsedPort < 0 || parsedPort > 65535) {
+          throw new IllegalArgumentException(
+              "Invalid port number " + parsedPort + " provided for block: " + blockName);
+        }
+        parsedPorts.add(parsedPort);
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException(
+            "Invalid port format provided for block: " + blockName, e);
+      }
+    }
+    return parsedPorts;
+  }
+
+  /**
+   * Validates a given URL against the blocking criteria.
+   *
+   * <p>If the URL's port matches any of the specified port numbers, a {@link
+   * io.camunda.connector.api.error.ConnectorInputException} is thrown, indicating that the URL is
+   * blocked.
+   *
+   * @param url The URL to validate, encapsulated as a {@link GenericUrl}.
+   * @throws io.camunda.connector.api.error.ConnectorInputException if the URL matches the block
+   *     conditions.
+   */
+  public void validate(GenericUrl url) {
+    if (blockedPorts.contains(url.getPort())) {
+      BlocklistExceptionHelper.throwBlocklistException("port", blockName);
+    }
+  }
+
+  public String value() {
+    return value;
+  }
+
+  public String blockName() {
+    return blockName;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) return true;
+    if (obj == null || obj.getClass() != this.getClass()) return false;
+    var that = (PortBlock) obj;
+    return Objects.equals(this.value, that.value) && Objects.equals(this.blockName, that.blockName);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(value, blockName);
+  }
+
+  @Override
+  public String toString() {
+    return "PortBlock[" + "value=" + value + ", " + "blockName=" + blockName + ']';
+  }
```

Comment:
Could you use records for the block implementations?

Reply:
You can have a constructor also for records if thats what you want: https://www.baeldung.com/java-records-custom-constructor

Reply:
First of all we didnt specify we need regex so we are overdoing this a little bit 😄 

You could also require to have a pattern in the constructor:  MyRecord(String name, Pattern regex) and create a static factory method that creates the instance.

Reply:
Why would you need the `String value` in the record?